scootapi/server/runJob.go

```diff
@@ -0,0 +1,93 @@
+package server
+
+import (
+	"fmt"
+	"github.com/nu7hatch/gouuid"
+	"github.com/scootdev/scoot/sched"
+	"github.com/scootdev/scoot/sched/scheduler"
+	"github.com/scootdev/scoot/scootapi/gen-go/scoot"
+)
+
+// Implementation of the RunJob API
+func runJob(scheduler scheduler.Scheduler, def *scoot.JobDefinition) (*scoot.JobId, error) {
+
+	jobDef, err := thriftJobToScoot(def)
+	// TODO: change to return scoot.NewInvalidRequest()
+	if err != nil {
+		return nil, err
+	}
+
+	err = validateJob(jobDef)
+	if err != nil {
+		return nil, err
+	}
+
+	job := sched.Job{
+		Id:  generateJobId(),
+		Def: jobDef,
+	}
+
+	err = scheduler.ScheduleJob(job)
```

Comment:
Let me think about it, i think this is probably ok.  I just want to let it simmer.  Will add in a subsequent code review if that's the case.  Will be an easy refactor